src/python/pants/engine/target.py

```diff
@@ -1502,13 +1502,17 @@ class SourcesField(AsyncFieldMixin, Field):
         files. The default is no limit on the number of source files.
     - `uses_source_roots` -- Whether the concept of "source root" pertains to the source files
         referenced by this field.
+    - `default` -- A default value for this field.
+    - `default_glob_match_error_behavior` -- Override glob match error behavior when using the
+        default value.
```

Comment:
So the idea with this is to have a default glob, and to match it if people happen to create a file with the right name, but to do something else if the file doesn't exist?

I know that that was discussed on the other PR, but it seems pretty error prone... if you are intending to use the file, but you get the name wrong, or accidentally delete it, there is nothing in the BUILD file signalling that intent, and no warning that the filename is incorrect. But I suppose that you have other validation (an error message that says something like "you must either have an existing source file or this other field"), so maybe that's fine.

So maybe just expand the help here to suggest not using this unless you have other validation that the file exists?

Reply:
Precisely so. If you misplace the file, you'll still get an error later telling you there's no Dockerfile (in that case). 

Will amend the help text.

Reply:
Maybe mention that this should usually not be set?